internal/conf/jwk.go

```diff
@@ -0,0 +1,192 @@
+	PublicKey  jwk.Key `json:"public_key"`
+	PrivateKey jwk.Key `json:"private_key"`
+}
+
+// Decode implements the Decoder interface
+// which transforms the keys stored as der binary strings into jwks
+func (j *JwtKeysDecoder) Decode(value string) error {
+	data := make([]map[string]interface{}, 0)
+	if err := json.Unmarshal([]byte(value), &data); err != nil {
+		return err
+	}
+
+	config := JwtKeysDecoder{}
+	for _, key := range data {
+		bytes, err := json.Marshal(key)
+		if err != nil {
+			return err
+		}
+		privJwk, err := jwk.ParseKey(bytes)
+		if err != nil {
+			return err
+		}
+		pubJwk, err := jwk.PublicKeyOf(privJwk)
+		if err != nil {
+			return err
+		}
+
+		// all public keys will be used for signature verification
+		if pubJwk.KeyUsage() == "enc" {
+			if err := pubJwk.Set(jwk.KeyUsageKey, "sig"); err != nil {
+				return err
+			}
+		}
+		config[pubJwk.KeyID()] = JwkInfo{
+			PublicKey:  pubJwk,
+			PrivateKey: privJwk,
+		}
+	}
+	*j = config
+	return nil
+}
+
+func (j *JwtKeysDecoder) Validate() error {
+	// Validate performs _minimal_ checks if the data stored in the key are valid.
+	// By minimal, we mean that it does not check if the key is valid for use in
+	// cryptographic operations. For example, it does not check if an RSA key's
+	// `e` field is a valid exponent, or if the `n` field is a valid modulus.
+	// Instead, it checks for things such as the _presence_ of some required fields,
+	// or if certain keys' values are of particular length.
+	//
+	// Note that depending on th underlying key type, use of this method requires
+	// that multiple fields in the key are properly populated. For example, an EC
+	// key's "x", "y" fields cannot be validated unless the "crv" field is populated first.
+	for _, key := range *j {
+		if err := key.PrivateKey.Validate(); err != nil {
+			return err
+		}
+		// symmetric keys don't have public keys
+		if key.PublicKey != nil {
+			if err := key.PublicKey.Validate(); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func GetSigningJwk(config *JWTConfiguration) (*jwk.Key, error) {
+	if config.Keys == nil {
+		symmetricJwk, err := getSymmetricKey(config)
+		if err != nil {
+			return nil, err
+		}
+		return symmetricJwk, nil
+	}
+	for _, key := range config.Keys {
+		if key.PrivateKey.KeyUsage() == "enc" {
+			return &key.PrivateKey, nil
+		}
+	}
+	// return symmetric secret anyway to preserve existing behavior
+	symmetricJwk, err := getSymmetricKey(config)
+	if err != nil {
+		return nil, err
+	}
+	return symmetricJwk, nil
+}
+
+func GetSigningKey(k *jwk.Key) (any, error) {
+	var key any
+	switch (*k).KeyType() {
+	case jwa.OctetSeq:
+		key = []byte{}
+	case jwa.EC:
+		key = ecdh.PrivateKey{}
+	case jwa.RSA:
+		key = rsa.PrivateKey{}
+	case jwa.OKP:
+		// OKP is used for EdDSA keys
+		key = ed25519.PrivateKey{}
+	case jwa.InvalidKeyType:
+		return nil, jwt.ErrInvalidKeyType
+	}
+	if err := (*k).Raw(&key); err != nil {
+		return nil, err
+	}
+	return key, nil
```

Comment:
I'm not sure why you need the `switch` above. You can just do `.Raw(&key)` and then return the key variable.

Reply:
hmm the [docs](https://github.com/lestrrat-go/jwx/blob/ce127d8a5a54a74d95df49fd2b9869fac177f891/jwk/interface_gen.go#L75) say that it's slower if the type isn't specified

Reply:
Smells like premature optimization. They probably have the exact same switch statement internally.

Reply:
yeah it's strange that their documentation recommended that... wrote a benchmark test and it's fast without the type assertions